File: Vincent OS/Calc.vb
﻿Public Class Calc
    ''Interface à Retravailler
    Dim Firstnum As Decimal
    Dim Secondnum As Decimal
    Dim Operation As Integer
    Dim Operator_Selector As Boolean = False

    Private Sub Button1_Click(sender As Object, e As EventArgs) Handles Button1.Click
        If TextBox1.Text <> "0" Then
            TextBox1.Text += "1"
        Else
            TextBox1.Text = "1"
        End If
    End Sub

    Private Sub Button2_Click(sender As Object, e As EventArgs) Handles Button2.Click
        If TextBox1.Text <> "0" Then
            TextBox1.Text += "2"
        Else
            TextBox1.Text = "2"
        End If
    End Sub

    Private Sub Button3_Click(sender As Object, e As EventArgs) Handles Button3.Click
        If TextBox1.Text <> "0" Then
            TextBox1.Text += "3"
        Else
            TextBox1.Text = "3"
        End If
    End Sub

    Private Sub Button4_Click(sender As Object, e As EventArgs) Handles Button4.Click
        If TextBox1.Text <> "0" Then
            TextBox1.Text += "4"
        Else
            TextBox1.Text = "4"
        End If
    End Sub

    Private Sub Button5_Click(sender As Object, e As EventArgs) Handles Button5.Click
        If TextBox1.Text <> "0" Then
            TextBox1.Text += "5"
        Else
            TextBox1.Text = "5"
        End If
    End Sub

    Private Sub Button6_Click(sender As Object, e As EventArgs) Handles Button6.Click
        If TextBox1.Text <> "0" Then
            TextBox1.Text += "6"
        Else
            TextBox1.Text = "6"
        End If
    End Sub

    Private Sub Button7_Click(sender As Object, e As EventArgs) Handles Button7.Click
        If TextBox1.Text <> "0" Then
            TextBox1.Text += "7"
        Else
            TextBox1.Text = "7"
        End If
    End Sub

    Private Sub Button8_Click(sender As Object, e As EventArgs) Handles Button8.Click
        If TextBox1.Text <> "0" Then
            TextBox1.Text += "8"
        Else
            TextBox1.Text = "8"
        End If
    End Sub

    Private Sub Button9_Click(sender As Object, e As EventArgs) Handles Button9.Click
        If TextBox1.Text <> "0" Then
            TextBox1.Text += "9"
        Else
            TextBox1.Text = "9"
        End If
    End Sub

    Private Sub Button10_Click(sender As Object, e As EventArgs) Handles Button10.Click
        If Not (TextBox1.Text.Contains(".")) Then
            TextBox1.Text += "."
        End If
    End Sub

    Private Sub Button11_Click(sender As Object, e As EventArgs) Handles Button11.Click
        If TextBox1.Text <> "0" Then
            TextBox1.Text += "0"
        End If
    End Sub

    Private Sub Button17_Click(sender As Object, e As EventArgs) Handles Button17.Click
        TextBox1.Text = "0"
    End Sub
    '' Système opération
    Private Sub Button13_Click(sender As Object, e As EventArgs) Handles Button13.Click
        Firstnum = TextBox1.Text
        TextBox1.Text = "0"
        Operator_Selector = True
        Operation = 1
    End Sub

    Private Sub Button14_Click(sender As Object, e As EventArgs) Handles Button14.Click
        Firstnum = TextBox1.Text
        TextBox1.Text = "0"
        Operator_Selector = True
        Operation = 2
    End Sub

    Private Sub Button15_Click(sender As Object, e As EventArgs) Handles Button15.Click
        Firstnum = TextBox1.Text
        TextBox1.Text = "0"
        Operator_Selector = True
        Operation = 3
    End Sub

    Private Sub Button16_Click(sender As Object, e As EventArgs) Handles Button16.Click
        Firstnum = TextBox1.Text
        TextBox1.Text = "0"
        Operator_Selector = True
        Operation = 4
    End Sub

    Private Sub Button12_Click(sender As Object, e As EventArgs) Handles Button12.Click
        If Operator_Selector = True Then
            Secondnum = TextBox1.Text
            If Operation = 1 Then
                TextBox1.Text = Firstnum + Secondnum
            ElseIf Operation = 2 Then
                TextBox1.Text = Firstnum - Secondnum
            ElseIf Operation = 3 Then
                TextBox1.Text = Firstnum * Secondnum
            Else
                If Secondnum = 0 Then
                    TextBox1.Text = "Div par zéro impossible !"
                Else
                    TextBox1.Text = Firstnum / Secondnum
                End If
            End If
            Operator_Selector = False
        End If
    End Sub
End Class

File: src/Vincent OS/Calc.cs
using System;
using System.Windows.Forms;
using Microsoft.VisualBasic.CompilerServices;

namespace Vincent_OS
{
    public partial class Calc
    {
        // 'Interface à Retravailler
        private decimal Firstnum;
        private decimal Secondnum;
        private int Operation;
        private bool Operator_Selector = false;

        public Calc()
        {
            InitializeComponent();
        }

        private void Calc_Load(object sender, EventArgs e)
        {
            if (My.MyProject.Forms.Bureau.Visible == true)
            {
                My.MyProject.Forms.Bureau.Button26.Visible = true;
            }
            else if (My.MyProject.Forms.Bureau2.Visible == true)
            {
                My.MyProject.Forms.Bureau2.Button26.Visible = true;
            }
        }

        private void Button1_Click(object sender, EventArgs e)
        {
            if (TextBox1.Text != "0")
            {
                TextBox1.Text += "1";
            }
            else
            {
                TextBox1.Text = "1";
            }
        }

        private void Button2_Click(object sender, EventArgs e)
        {
            if (TextBox1.Text != "0")
            {
                TextBox1.Text += "2";
            }
            else
            {
                TextBox1.Text = "2";
            }
        }

        private void Button3_Click(object sender, EventArgs e)
        {
            if (TextBox1.Text != "0")
            {
                TextBox1.Text += "3";
            }
            else
            {
                TextBox1.Text = "3";
            }
        }

        private void Button4_Click(object sender, EventArgs e)
        {
            if (TextBox1.Text != "0")
            {
                TextBox1.Text += "4";
            }
            else
            {
                TextBox1.Text = "4";
            }
        }

        private void Button5_Click(object sender, EventArgs e)
        {
            if (TextBox1.Text != "0")
            {
                TextBox1.Text += "5";
            }
            else
            {
                TextBox1.Text = "5";
            }
        }

        private void Button6_Click(object sender, EventArgs e)
        {
            if (TextBox1.Text != "0")
            {
                TextBox1.Text += "6";
            }
            else
            {
                TextBox1.Text = "6";
            }
        }

        private void Button7_Click(object sender, EventArgs e)
        {
            if (TextBox1.Text != "0")
            {
                TextBox1.Text += "7";
            }
            else
            {
                TextBox1.Text = "7";
            }
        }

        private void Button8_Click(object sender, EventArgs e)
        {
            if (TextBox1.Text != "0")
            {
                TextBox1.Text += "8";
            }
            else
            {
                TextBox1.Text = "8";
            }
        }

        private void Button9_Click(object sender, EventArgs e)
        {
            if (TextBox1.Text != "0")
            {
                TextBox1.Text += "9";
            }
            else
            {
                TextBox1.Text = "9";
            }
        }

        private void Button10_Click(object sender, EventArgs e)
        {
            if (!TextBox1.Text.Contains(","))
            {
                TextBox1.Text += ",";
            }
        }

        private void Button11_Click(object sender, EventArgs e)
        {
            if (TextBox1.Text != "0")
            {
                TextBox1.Text += "0";
            }
        }

        private void Button17_Click(object sender, EventArgs e)
        {
            TextBox1.Text = "0";
        }
        // ' Système opération
        private void Button13_Click(object sender, EventArgs e)
        {
            Firstnum = Conversions.ToDecimal(TextBox1.Text);
            TextBox1.Text = "0";
            Operator_Selector = true;
            Operation = 1;
        }

        private void Button14_Click(object sender, EventArgs e)
        {
            Firstnum = Conversions.ToDecimal(TextBox1.Text);
            TextBox1.Text = "0";
            Operator_Selector = true;
            Operation = 2;
        }

        private void Button15_Click(object sender, EventArgs e)
        {
            Firstnum = Conversions.ToDecimal(TextBox1.Text);
            TextBox1.Text = "0";
            Operator_Selector = true;
            Operation = 3;
        }

        private void Button16_Click(object sender, EventArgs e)
        {
            Firstnum = Conversions.ToDecimal(TextBox1.Text);
            TextBox1.Text = "0";
            Operator_Selector = true;
            Operation = 4;
        }

        private void Button12_Click(object sender, EventArgs e)
        {
            if (Operator_Selector == true)
            {
                Secondnum = Conversions.ToDecimal(TextBox1.Text);
                if (Operation == 1)
                {
                    TextBox1.Text = (Firstnum + Secondnum).ToString();
                }
                else if (Operation == 2)
                {
                    TextBox1.Text = (Firstnum - Secondnum).ToString();
                }
                else if (Operation == 3)
                {
                    TextBox1.Text = (Firstnum * Secondnum).ToString();
                }
                else if (Secondnum == 0m)
                {
                    TextBox1.Text = "Div par zéro impossible !";
                }
                else
                {
                    TextBox1.Text = (Firstnum / Secondnum).ToString();
                }
                Operator_Selector = false;
            }
        }

        private void Calc_FormClosed(object sender, FormClosedEventArgs e)
        {
            if (My.MyProject.Forms.Bureau.Visible == true)
            {
                My.MyProject.Forms.Bureau.Button26.Visible = false;
            }
            else if (My.MyProject.Forms.Bureau2.Visible == true)
            {
                My.MyProject.Forms.Bureau2.Button26.Visible = false;
            }
        }
    }
}